src/EventTickets/Migrations/CreateEventsTable.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Give\EventTickets\Migrations;
+
+use Give\Framework\Database\DB;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+/**
+ * @unreleased
+ */
+class CreateEventsTable extends Migration {
+    /**
+     * @inheritdoc
+     */
+    public static function id() {
+        return 'give-events-create-events-table';
+    }
+
+    public static function title() {
+        return 'Create give_events table';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function timestamp() {
+        return strtotime( '2022-01-29 00:00:00' );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function run() {
+        global $wpdb;
+
+        $table   = $wpdb->give_events;
+        $charset = DB::get_charset_collate();
+
+        $sql = "CREATE TABLE $table (
+			id INT UNSIGNED NOT NULL,
+			form_id INT NULL,
+			description TEXT NULL,
+			start_datetime DATETIME NULL,
+			end_datetime DATETIME NULL,
+			ticket_close_datetime DATETIME NULL,
+			PRIMARY KEY  (id)
+		) $charset";
```

Comment:
I believe we are missing columns like `created_at` and `updated_at` (same in the other new tables).

Jeffrey has added a Timezone field. Will we include it in the formatted DateTime object before saving? We will need to retrieve that information to display it in the field template.

Reply:
I'll add the `created_at` and `updated_at` columns for the Tickets and Ticket Types tables (Events inherits this from the CPT).

Re: Timezone: I'm unsure how we should handle that and would rather add that later when we have a chance to better understand the best practice for handling timezones.

Reply:
Resolved, see [a053c51](https://github.com/impress-org/givewp/pull/7213/commits/a053c51bf765277bfa85ce38cde81cfcd64d23cd).